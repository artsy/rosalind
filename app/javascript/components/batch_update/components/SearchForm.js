import React from 'react'
import styled from 'styled-components'
import CurrentCriteria from './CurrentCriteria'
import { CreatedAfterDateInput, CreatedBeforeDateInput } from './DateInput'
import TextInput from './TextInput'
import {
  ArtistAutosuggest,
  AttributionClassAutosuggest,
  FairAutosuggest,
  GeneAutosuggest,
  PartnerAutosuggest,
  TagAutosuggest
} from './Autosuggest'
import FilterOptions from './FilterOptions'
import Button from '@artsy/reaction/dist/Components/Buttons/Default'
import InvertedButton from '@artsy/reaction/dist/Components/Buttons/Inverted'
import { Link } from './Links'
import { SitesConsumer } from '../SitesContext'

const fullWidth = `
  margin: 0;
  width: 100%;
`
const FullWidthButton = styled(Button)`
  ${fullWidth}
`

const FullWidthInvertedButton = styled(InvertedButton)`
  ${fullWidth}
`

class SearchForm extends React.Component {
  constructor (props) {
    super(props)
    this.maybeRenderEditButton = this.maybeRenderEditButton.bind(this)
  }

  maybeRenderEditButton () {
    const {
      artworksCount,
      selectedArtworksCount,
      selectedArtworkIds,
      onOpenBatchUpdate
    } = this.props
    if (artworksCount === 0) {
      return null
    } else if (selectedArtworksCount === 0) {
      return <FullWidthButton disabled>Edit Artworks</FullWidthButton>
    } else {
      return (
        <React.Fragment>
          <FullWidthInvertedButton onClick={onOpenBatchUpdate}>
            Edit Artworks
          </FullWidthInvertedButton>
          <HelixLink selectedArtworkIds={selectedArtworkIds} />
        </React.Fragment>

      )
    }
  }

  render () {
    const {
      artists,
      attributionClass,
      clearState,
      createdAfterDate,
      createdBeforeDate,
      fair,
      genes,
      keywords,
      onRemoveKeyword,
      onRemoveGene,
      onRemoveTag,
      onRemoveArtist,
      partner,
      tags
    } = this.props

    const { onAddKeyword, onAddGene, onAddTag, onAddArtist, updateState } = this.props

    const { genomedFilter, publishedFilter } = this.props

    return (
      <div className={this.props.className}>
        <CurrentCriteria
          artists={artists}
          attributionClass={attributionClass}
          clearState={clearState}
          createdAfterDate={createdAfterDate}
          createdBeforeDate={createdBeforeDate}
          fair={fair}
          genes={genes}
          keywords={keywords}
          onRemoveKeyword={onRemoveKeyword}
          onRemoveGene={onRemoveGene}
          onRemoveTag={onRemoveTag}
          onRemoveArtist={onRemoveArtist}
          partner={partner}
          tags={tags}
          updateState={updateState}
        />

        <TextInput placeholder='Add a keyword' onEnter={onAddKeyword} />
        <GeneAutosuggest placeholder='Add a gene' onSelectGene={onAddGene} />
        <TagAutosuggest placeholder='Add a tag' onSelectTag={onAddTag} />
        <ArtistAutosuggest
          placeholder='Add an artist'
          onSelectArtist={onAddArtist}
        />
        {partner === null && <PartnerAutosuggest updateState={updateState} />}
        {fair === null && <FairAutosuggest updateState={updateState} />}
        {attributionClass === null && <AttributionClassAutosuggest updateState={updateState} />}
        {createdAfterDate === null && (
          <CreatedAfterDateInput
            updateState={updateState}
            createdAfterDate={createdAfterDate}
          />
        )}
        {createdBeforeDate === null && (
          <CreatedBeforeDateInput
            updateState={updateState}
            createdBeforeDate={createdBeforeDate}
          />
        )}

        <FilterOptions
          genomedFilter={genomedFilter}
          publishedFilter={publishedFilter}
          updateState={updateState}
        />

        {this.maybeRenderEditButton()}
      </div>
    )
  }
}

const HelixLink = ({selectedArtworkIds}) => {
  return (
    <SitesConsumer>
      {
        sites => {
          const href = `${sites.helix}/genome/artworks?artwork_ids=${selectedArtworkIds.join(',')}`
          return (
            <_Link target='_blank' href={href}>
              Open selected works in Helix
            </_Link>
          )
        }
      }
    </SitesConsumer>
  )
}

const _Link = styled(Link)`
  display: block;
  margin-top: 1em;
`

/* default styled component */

const StyledSearchForm = styled(SearchForm)``

export default StyledSearchForm
