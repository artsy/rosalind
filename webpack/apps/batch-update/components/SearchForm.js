import React from 'react'
import styled from 'styled-components'
import CurrentCriteria from './CurrentCriteria'
import {
  CreatedAfterDateInput,
  CreatedBeforeDateInput
} from './DateInput'
import {
  FairAutosuggest,
  GeneAutosuggest,
  PartnerAutosuggest,
  TagAutosuggest
} from './Autosuggest'
import FilterOptions from './FilterOptions'
import { Button } from './Buttons'

class SearchForm extends React.Component {
  constructor (props) {
    super(props)
    this.maybeRenderEditButton = this.maybeRenderEditButton.bind(this)
  }

  maybeRenderEditButton () {
    const { artworksCount, selectedArtworksCount, onOpenBatchUpdate } = this.props
    if (artworksCount === 0) {
      return null
    } else if (selectedArtworksCount === 0) {
      return <Button fullWidth disabled>Edit Artworks</Button>
    } else {
      return <Button fullWidth primary onClick={onOpenBatchUpdate}> Edit Artworks </Button>
    }
  }

  render () {
    const {
      clearState,
      createdAfterDate,
      createdBeforeDate,
      fair,
      genes,
      onRemoveGene,
      onRemoveTag,
      partner,
      tags
    } = this.props

    const {
      onAddGene,
      onAddTag,
      updateState
    } = this.props

    const {
      genomedFilter,
      publishedFilter
    } = this.props

    return (
      <div className={this.props.className}>
        <CurrentCriteria
          clearState={clearState}
          createdAfterDate={createdAfterDate}
          createdBeforeDate={createdBeforeDate}
          fair={fair}
          genes={genes}
          onRemoveGene={onRemoveGene}
          onRemoveTag={onRemoveTag}
          partner={partner}
          tags={tags}
          updateState={updateState}
          />

        <GeneAutosuggest placeholder='Add a gene' onSelectGene={onAddGene} />
        <TagAutosuggest placeholder='Add a tag' onSelectTag={onAddTag} />
        {partner === null && <PartnerAutosuggest updateState={updateState} />}
        {fair === null && <FairAutosuggest updateState={updateState} />}
        {
          createdAfterDate === null &&
            <CreatedAfterDateInput
              updateState={updateState}
              createdAfterDate={createdAfterDate}
              />
        }
        {
          createdBeforeDate === null &&
            <CreatedBeforeDateInput
              updateState={updateState}
              createdBeforeDate={createdBeforeDate}
              />
        }

        <FilterOptions
          genomedFilter={genomedFilter}
          publishedFilter={publishedFilter}
          updateState={updateState}
          />

        {this.maybeRenderEditButton()}
      </div>
    )
  }
}

/* default styled component */

const StyledSearchForm = styled(SearchForm)`
`

export default StyledSearchForm
