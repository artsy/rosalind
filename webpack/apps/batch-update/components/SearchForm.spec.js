import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { mount } from 'enzyme'
import moment from 'moment'
import SearchForm from './SearchForm'

let props

beforeEach(() => {
  props = {
    artworksCount: 0,
    createdAfterDate: null,
    createdBeforeDate: null,
    fair: null,
    genes: [],
    genomedFilter: 'SHOW_ALL',
    onAddGene: jest.fn(),
    onAddTag: jest.fn(),
    onClearFair: jest.fn(),
    onClearPartner: jest.fn(),
    onRemoveGene: jest.fn(),
    onRemoveTag: jest.fn(),
    partner: null,
    publishedFilter: 'SHOW_ALL',
    selectedArtworksCount: 0,
    tags: [],
    updateStateFor: jest.fn()
  }
})

it('renders correctly', () => {
  const rendered = renderer.create(<SearchForm {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render partner autosuggest if partner is already selected', () => {
  const partner = { id: 'nice-gallery', name: 'Nice Gallery' }
  Object.assign(props, { partner })
  const rendered = renderer.create(<SearchForm {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render fair autosuggest if fair is already selected', () => {
  const fair = { id: 'nice-fair', name: 'Nice Fair' }
  Object.assign(props, { fair })
  const rendered = renderer.create(<SearchForm {...props} />)
  const tree = rendered.toJSON()
  expect(tree).toMatchSnapshot()
})

it('does not render createdAfterDate input if createdAfterDate is already entered', () => {
  const createdAfterDate = moment('2020-01-01T12:00:00-00:00').utc().format()
  Object.assign(props, { createdAfterDate })

  const searchForm = mount(<SearchForm {...props} />)
  const currentCreatedAfterDate = searchForm.find('SelectedCreatedAfterDate')
  const createdAfterDateInput = searchForm.find('CreatedAfterDateInput')

  expect(currentCreatedAfterDate.length).toEqual(1)
  expect(createdAfterDateInput.length).toEqual(0)
})

it('does not render createdBeforeDate input if createdBeforeDate is already entered', () => {
  const createdBeforeDate = moment('2020-01-01T12:00:00-00:00').utc().format()
  Object.assign(props, { createdBeforeDate })

  const searchForm = mount(<SearchForm {...props} />)
  const currentCreatedBeforeDate = searchForm.find('SelectedCreatedBeforeDate')
  const createdBeforeDateInput = searchForm.find('CreatedBeforeDateInput')

  expect(currentCreatedBeforeDate.length).toEqual(1)
  expect(createdBeforeDateInput.length).toEqual(0)
})

describe('"edit artworks" button', () => {
  it('does not render an edit button if there are no artworks', () => {
    Object.assign(props, { artworksCount: 0 })
    const rendered = renderer.create(<SearchForm {...props} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders a disabled edit button if there are artworks, but none selected', () => {
    Object.assign(props, { artworksCount: 100, selectedArtworksCount: 0 })
    const rendered = renderer.create(<SearchForm {...props} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders an enabled edit button if there are selected artworks', () => {
    Object.assign(props, { artworksCount: 100, selectedArtworksCount: 1 })
    const rendered = renderer.create(<SearchForm {...props} />)
    const tree = rendered.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
