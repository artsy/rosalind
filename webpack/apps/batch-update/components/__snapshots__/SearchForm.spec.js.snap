exports[`"edit artworks" button does not render an edit button if there are no artworks 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq" />
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-partner-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a partner"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-partner-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-fair-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a fair"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-fair-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
</div>
`;

exports[`"edit artworks" button renders a disabled edit button if there are artworks, but none selected 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq" />
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-partner-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a partner"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-partner-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-fair-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a fair"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-fair-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
  <button
    className="dsgyEb"
    disabled={true}>
    Edit Artworks
  </button>
</div>
`;

exports[`"edit artworks" button renders an enabled edit button if there are selected artworks 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq" />
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-partner-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a partner"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-partner-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-fair-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a fair"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-fair-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
  <button
    className="iAICvi"
    onClick={undefined}>
     Edit Artworks 
  </button>
</div>
`;

exports[`test does not render fair autosuggest if fair is already selected 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq">
    <div>
      <h2>
        Fair
      </h2>
      <div
        className="crrNnH">
        Nice Fair
        <a
          className="remove"
          href="#"
          onClick={[Function]}>
          ✕
        </a>
      </div>
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-partner-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a partner"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-partner-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
</div>
`;

exports[`test does not render partner autosuggest if partner is already selected 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq">
    <div>
      <h2>
        Partner
      </h2>
      <div
        className="crrNnH">
        Nice Gallery
        <a
          className="remove"
          href="#"
          onClick={[Function]}>
          ✕
        </a>
      </div>
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-fair-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a fair"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-fair-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
</div>
`;

exports[`test renders correctly 1`] = `
<div
  className="foqgVb">
  <div
    className="cMTqqq" />
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-gene-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a gene"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-gene-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-tag-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Add a tag"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-tag-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-partner-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a partner"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-partner-autosuggest" />
    </div>
  </div>
  <div
    className="GenericAutosuggest">
    <div
      className="react-autosuggest__container">
      <input
        aria-activedescendant={null}
        aria-autocomplete="list"
        aria-expanded={false}
        aria-haspopup={false}
        aria-owns="react-autowhatever-fair-autosuggest"
        autoComplete="off"
        className="react-autosuggest__input"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="Select a fair"
        role="combobox"
        type="text"
        value="" />
      <div
        className="react-autosuggest__suggestions-container"
        id="react-autowhatever-fair-autosuggest" />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created after date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="iFTuzH">
    <input
      onChange={[Function]}
      onKeyPress={[Function]}
      placeholder="Created before date"
      type="text"
      value="" />
    <div
      className="parsed">
      <a
        href="#"
        onClick={[Function]} />
    </div>
  </div>
  <div
    className="gxGcSj">
    <div
      className="filter">
      <div>
        Published?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Published
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not published
      </a>
    </div>
    <div
      className="filter">
      <div>
        Genomed?
      </div>
      <a
        className="active"
        href="#"
        onClick={[Function]}>
        All
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Genomed
      </a>
      <a
        className={null}
        href="#"
        onClick={[Function]}>
        Not genomed
      </a>
    </div>
  </div>
</div>
`;
